### PR TITLE
Multipart parser support

### PR DESCRIPTION
**Pull request by sunharis**
As part of HMC performance improvement, this commit implements
multiple configfile upload with a single http request

Tested by:
1. Multiple file upload
curl -k -H "X-Auth-Token: $bmc_token" -H "Content-Type: multipart/form-data"
-X POST https://${bmc}/ibm/v1/Host/ConfigFiles -F "testfile1=@<path>;type=application/octet-stream"
-F "testfile2=@<path>;type=application/octet-stream"
2. Tested the single file upload for sanity

### CODE REVIEW

**Comment by sunharis** [comment]
Multipart parser is from : https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/34974

The commit d46c7b6 implements https://github.com/ibm-openbmc/dev/issues/3351 . There is no upstream commit for this command implementation as this is /ibm/v1 path.

Below command works for uploading the multiple save-area files at one go.

_curl -k -H "X-Auth-Token: $bmc_token" -H "Content-Type: multipart/form-data"
-X POST https://${bmc}/ibm/v1/Host/ConfigFiles -F "testfile1=@<path>;type=application/octet-stream"
-F "testfile2=@<path>;type=application/octet-stream"_

File: include/ibm/management_console_rest.hpp [hunk 1]
```diff
@@ -238,6 +173,161 @@ inline void handleFilePut(const crow::Request& req,
...
+    {
+        asyncResp->res.result(boost::beast::http::status::not_acceptable);
+        asyncResp->res.jsonValue["Description"] = contentNotAcceptableMsg;
+        return;
+    }
+
+    BMCWEB_LOG_DEBUG
+        << "handleIbmPut: Request to create/update the save-area file";
+    std::string_view path =
+        "/var/lib/bmcweb/ibm-management-console/configfiles";
```
Comment (asmithakarun) [inline include/ibm/management_console_rest.hpp:205]: Can we have a const variable initialized to this on top and use it elsewhere?

**Comment by asmithakarun** [comment]
Can we fix the CI failure?

**Review by gtmills — CHANGES_REQUESTED** [review]
We need to discuss more if we want this.

**Comment by gtmills** [comment]
Can we get this rebased and refreshed to pick up the latest https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/34974 ?

**Comment by sunharis** [comment]
PR is not yet ready

**Comment by sunharis** [comment]
https://github.com/ibm-openbmc/bmcweb/pull/180 is created

**Pull request closed without merging**